artman/cli/configure.py

```diff
@@ -46,107 +51,116 @@ def configure(log_level=logging.INFO):
-        logger.info('First, we need to know where you store most code on your '
-                    'local machine.')
-        logger.info('Other paths (example: toolkit) will derive from this, '
-                    'but most are individually configurable.')
-        logger.info('The use of ${REPOROOT} in GAPIC YAMLs will point here.')
-        logger.info('Note: Use of ~ is fine here.')
-        answer['reporoot'] = six.moves.input('Local code path: ')
-        answer['reporoot'] = answer['reporoot'].rstrip('/').strip()
-
-    # Set up dependent directories.
-    reporoot = answer['reporoot']
-    for dep in ('api-client-staging', 'googleapis', 'toolkit'):
+    answer = LocalConfig()
+
+    # Ask the user for a local toolkit location.
+    while not answer.toolkit:
         location = six.moves.input(
-            'Path for {0} (default: {1}/{0}): '.format(dep, reporoot)
-        ).rstrip('/').strip()
+            'Where is your local toolkit? Please provide an absolute path: ')
         if location:
-            answer[dep.replace('-', '_')] = location
+            answer.toolkit = location
 
     # Done; return the answer.
     return answer
 
 
-def _configure_publish(publish=None):
-    """Determine and return the default publisher.
-
-    Args:
-        publish (str): The current default publisher (may be None).
-
-    Returns:
-        str: The new default publisher.
-    """
-    # Set up publishing defaults.
-    logger.info('Where do you want to publish code by default?')
-    logger.info('The common valid options are "github" and "local".')
-    publish = six.moves.input('Default publisher: ').lower()
-    try:
-        importlib.import_module('artman.tasks.publish.%s' % publish)
-        return publish
-    except ImportError:
-        logger.error('Invalid publisher.')
-        return _configure_publish()
-
-
-def _configure_github(github):
-    """Determine and return the GitHub configuration.
-
-    Args:
-        github (dict): The current GitHub configuration.
+def _configure_github_config():
+    """Determine and return artman user GitHub config.
 
     Returns:
         dict: The new GitHub configuration.
     """
-    answer = copy(github)
+    answer = GitHubConfig()
     logger.info('Since you intend to publish to GitHub, you need to '
                 'supply credentials.')
     logger.info('Create an access token at: '
                 'https://github.com/settings/tokens')
     logger.info('It needs the "repo" scope and nothing else.')
-    while not answer.get('username'):
-        answer['username'] = six.moves.input('GitHub username: ')
-    while not answer.get('token'):
-        answer['token'] = getpass.getpass('GitHub token (input is hidden): ')
+    while not answer.username:
+        answer.username = six.moves.input('GitHub username: ')
+    while not answer.token:
+        answer.token = getpass.getpass('GitHub token (input is hidden): ')
+
     return answer
+
+def _convert_json(d):
```

Comment:
Why do we need to covert proto names to snake_case?

Reply:
Good catch. Not necessary.